### PR TITLE
fuzz: add fuzzers for calendar & time expressions

### PR DESCRIPTION
**Pull request by mrc0mmand**
Another two smaller fuzzers (I promise I'll get to the bigger ones soon™) for calendar & time expressions (https://github.com/systemd/systemd/pull/11960#issuecomment-471630110).

### CODE REVIEW

File: src/fuzz/fuzz-calendarspec.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        _cleanup_(calendar_spec_freep) CalendarSpec *cspec = NULL;
+        _cleanup_free_ char *str = NULL, *p = NULL;
+
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        str = malloc0(sizeof(*data) * (size + 1));
+        assert_se(str != NULL);
+        memcpy_safe(str, data, size);
+        str[size] = '\0';
```
Comment (keszybz) [inline src/fuzz/fuzz-calendarspec.c]: Please use `memdup_suffix0` instead.

File: src/fuzz/fuzz-calendarspec.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        (void) calendar_spec_from_string(str, &cspec);
+        if (cspec) {
```
Comment (keszybz) [inline src/fuzz/fuzz-calendarspec.c]: Maybe it'd be better to use `if (calendar_spec_from_string() >= 0)` ? This tests that `cspec` is set on non-failure.

File: src/fuzz/fuzz-time-util.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#include "util.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        _cleanup_free_ char *str = NULL;
+        usec_t usec;
+
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        str = malloc0(sizeof(*data) * (size + 1));
```
Comment (keszybz) [inline src/fuzz/fuzz-time-util.c]: As above.

File: src/fuzz/fuzz-calendarspec.c (hunk 1 above)
Comment (mrc0mmand) [reply src/fuzz/fuzz-calendarspec.c]: I suspected there's a function dedicated just for that, will do, thanks!

File: src/fuzz/fuzz-calendarspec.c (hunk 2 above)
Comment (mrc0mmand) [reply src/fuzz/fuzz-calendarspec.c]: Definitely makes sense.

File: src/fuzz/fuzz-calendarspec.c (hunk 1 above)
Comment (mrc0mmand) [reply src/fuzz/fuzz-calendarspec.c]: Also, is there any particular reason why `memcpy_safe` (and probably `memcmp*/memset*` as well)  is defined in `basic/util.h` whereas the rest of the mem-related functions is in `basic/alloc-util.h`?

Comment (keszybz) [reply src/fuzz/fuzz-calendarspec.c]: A long time ago everything was in util.h, and some things were moved out. It's all a bit of a mess.

**Pull request merged**